manager/allocator/network/driver/doc.go

```diff
@@ -0,0 +1,11 @@
+package driver
+
+// The drivers package contains the Allocator in charge of setting a Network's
+// Driver state. Before a network can be used, if it depends on a global-scoped
+// driver, it must be allocated on that driver. This package provides the code
+// that handles that state allocation.
```

Comment:
Would it make sense to say, as in the `network/doc.go` doc, that allocations and restores mutate the objects passed to it to be committed by a caller further up the stack?

Reply:
Added some comments hopefully explaining this.